core/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -338,7 +344,18 @@ public ActionModule(boolean ingestEnabled, boolean transportClient, Settings set
         autoCreateIndex = transportClient ? null : new AutoCreateIndex(settings, clusterSettings, resolver);
         destructiveOperations = new DestructiveOperations(settings, clusterSettings);
         Set<String> headers = actionPlugins.stream().flatMap(p -> p.getRestHeaders().stream()).collect(Collectors.toSet());
-        restController = new RestController(settings, headers);
+        UnaryOperator<RestHandler> restWrapper = null;
+        for (ActionPlugin plugin : actionPlugins) {
+            UnaryOperator<RestHandler> newRestWrapper = plugin.getRestHandlerWrapper(threadPool.getThreadContext());
+            if (newRestWrapper != null) {
+                logger.debug("Using REST wrapper from plugin " + plugin.getClass().getName());
+                if (restWrapper != null) {
```

Comment:
I have no idea if there are plugins like this out there to be honest. I just see it as an odd change that you can have only one plugin installed that can do things before executing rest actions. And, my personal opinion, I don't see having a filter chain and multiple filters with ordering as over-engineering. But again, my own personal opinion and it doesn't mean that it should block the change, especially if I am the only one seeing it this way.